### ISSUE TITLE
romTool segfaults on older ROM files

### ISSUE THREAD

**Issue by jduerstock** [elliotnunn/toolboxtoolbox#1]
I ran your romTool against the collection of ROM files here: https://github.com/sentient06/MacROMan/tree/master/TestImages/New%20World%20ROM

It segfaults while examining the parcels for any versions older than 9.0.1.  I've never spent much time examining any of the parcels other than the "Mac OS ROM parcel" file, so I'm not quite sure where to start looking next.  Let me know what I can do to help.

Thanks.

**Comment by jesterspanic** [comment]
Sorry about that. I'm not really surprised honestly.
There was no effort made to work with those older versions.
Based on the intent of the project this is used with, I don't expect this code to ever be fixed for those older ROM's.

I did update the romTools for you with the most current version/source.
But I can't guarantee they will work in OS X

**Comment by jduerstock** [comment]
I was using it under Linux and it worked fine, except for the segfault part. :)  Is there a reason you switched them to MPW based projects?

Is there any particular area you're stuck on with your project?  I've been following it a little but I haven't paid particularly close attention.

**Comment by jesterspanic** [comment]
What you see here is part of a larger project and this project is basic tools for the larger project.

MPW is used because it runs on OS 9.2.2

Not stuck on anything at this point.

Have you been following on github only?

**Comment by jduerstock** [comment]
No, I've been reading on macos9lives too, although I don't understand the emulation hate there.

I'm not sure if you've seen this bit here, but I found it rather enlightening: https://groups.google.com/forum/#!msg/comp.sys.mac.programmer.help/fqdq59fxQ1g/uSy4DSDLeJcJ

**Comment by jduerstock** [comment]
I'm also a little confused as to why you'd merge the MPW version of romTools on top of (relatively well) functioning Unix source.  Wouldn't it be better hosted in a separate directory?

Regardless, thank you for all your work.  It is all very interesting to me.

**Comment by jesterspanic** [comment]
The merge was how I wrote it or should say updated/rewrote it. I will look into moving as that makes sense.
I had not seen that specific link. Pretty nice chunk of info.  There is a bunch of other info on that list as well that i have gone through.

Have you tested this against MPW and received the same segfault.

This is starting to become a more general conversation (though I am enjoying it).  If you would like to continue it you can message me at macos9lives. I go by nanopico on that forum.
